backend/cmm_helpers.ml

```diff
@@ -3689,7 +3689,12 @@ let make_unboxed_int32_array_payload dbg unboxed_int32_list =
     | [] -> Even, List.rev acc
     | a :: [] -> Odd, List.rev (a :: acc)
     | a :: b :: r ->
-      let i = Cop (Cor, [a; Cop (Clsl, [b; Cconst_int (32, dbg)], dbg)], dbg) in
+      let i =
+        Cop
+          ( Cor,
+            [zero_extend_32 dbg a; Cop (Clsl, [b; Cconst_int (32, dbg)], dbg)],
```

Comment:
I think this deserves a comment in the code, explaining that by default, `a` will be sign-extended (as are all Int32s), and we need it to be zero_extended instead for the `or` to be correct.